### PR TITLE
feat: blocks ssr

### PR DESCRIPTION
**Pull request by sandren**
This PR adds alternate initial rendering for block content.

### CODE REVIEW

File: apps/web/core/state/editor-store.tsx [hunk 1]
```diff
@@ -111,7 +112,7 @@ export function useEditorStore() {
         const html = markdownTriple ? markdownConverter.makeHtml(Value.stringValue(markdownTriple) || '') : '';
         /* SSR on custom react nodes doesn't seem to work out of the box at the moment */
         const isSSR = typeof window === 'undefined';
-        const json = isSSR ? { content: '' } : generateJSON(html, tiptapExtensions);
+        const json = isSSR ? generateServerJSON(html, tiptapExtensions) : generateJSON(html, tiptapExtensions);
```
Comment (baiirun) [inline apps/web/core/state/editor-store.tsx:115]: 🔥

File: apps/web/partials/editor/server-content.tsx [hunk 2]
```diff
@@ -0,0 +1,127 @@
+type ServerContentProps = {
+  content: any[];
```
Comment (baiirun) [inline apps/web/partials/editor/server-content.tsx:2]: Is there a decent way we can type this?

File: apps/web/partials/editor/server-content.tsx [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+      {content.map((block, index) => (
+        <Block key={index} block={block} />
+      ))}
+      <TrailingBreak />
+    </div>
+  );
+};
+
+type BlockProps = {
+  block: any;
```
Comment (baiirun) [inline apps/web/partials/editor/server-content.tsx:17]: Same question

File: apps/web/partials/editor/server-content.tsx [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+
+const Block = ({ block }: BlockProps) => {
+  switch (block.type) {
+    case 'paragraph': {
+      return (
+        <div className="react-renderer node-paragraph">
+          <div className="whitespace-normal">
+            <p>
+              {block.content.map((block: any, index: number) => (
+                <Block key={index} block={block} />
```
Comment (baiirun) [inline apps/web/partials/editor/server-content.tsx:28]: Been a minute since I've seen recursive components haha

Comment (sandren) [reply apps/web/partials/editor/server-content.tsx:28]: Recursion is always fun. 🙃

**Review by baiirun — APPROVED** [review]

**Pull request merged**